src/client.rs

```diff
@@ -240,6 +241,63 @@ impl HttpClientBuilder {
         self.configure(map)
     }
 
+    /// Set a default header to be passed with every request
+    ///
+    /// NOTE: In case there is an error in parsing the HeaderName or HeaderValue
+    /// the tuple is silently discarded.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use isahc::prelude::*;
+    /// #
+    /// let client = HttpClient::builder()
+    ///     .default_header("some-header", "some-value")
+    ///     .build()?;
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    pub fn default_header(
+        mut self,
+        key: impl TryInto<HeaderName>,
+        value: impl TryInto<HeaderValue>,
+    ) -> Self {
+        match key.try_into() {
+            Ok(key) => match value.try_into() {
+                Ok(value) => {
+                    if let Some(headers_map) = self.defaults.get_mut::<DefaultHeaderMap>() {
+                        headers_map.0.append(key, value);
+                    } else {
+                        let mut header = http::HeaderMap::with_capacity(1);
+                        header.insert(key, value);
+                        self.defaults.insert(DefaultHeaderMap(header));
+                    }
+                    self
+                }
+                Err(_e) => self,
+            },
+            Err(_e) => self,
```

Comment:
A common technique for builders accepting fallible parameters is to defer error handling until the end. Since `build()` returns a result, we could store the error here in the builder somehow and then check it first when calling the final `build()`. I recommend trying that approach here.

One way to do this would be to add an `error: Option<Error>` field to the builder.

Reply:
Thank you, this was exactly my confusion.